File: app/src/main/java/com/example/gmailish/data/entity/BlacklistEntity.java
package com.example.gmailish.data.entity;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.Date;

@Entity(tableName = "blacklist")
public class BlacklistEntity {
    @PrimaryKey @NonNull public String url;


    public Date createdAt;
    public Date updatedAt;

    public BlacklistEntity(@NonNull String url, Date createdAt, Date updatedAt) {
        this.url = url;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
    }
}

File: app/src/main/java/com/example/gmailish/data/entity/BlacklistEntity.kt
package com.example.gmailish.data.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date

@Entity(tableName = "blacklist")
data class BlacklistEntity(
    @PrimaryKey val url: String,
    val createdAt: Date? = null,
    val updatedAt: Date? = null
)